driver/bpf/fillers.h

```diff
@@ -6282,7 +6282,7 @@ FILLER(sys_dup3_x, true)
 	 */
 	val = bpf_syscall_get_argument(data, 2);
 	flags = dup3_flags_to_scap(val);
-	return bpf_push_u32_to_ring(data, flags);
+	return bpf_push_s32_to_ring(data, flags);
```

Comment:
@ecbadeaux could we double check this? I quickly checked on `dup3_flags_to_scap` and since we convert the original flags to scap flags the old uint32_t should be correct.

Instead what we need to do is not use `val` here and do something else, one way could be the following:

```
unsigned long val;
unsigned long retval;
int flags;
unsigned long res;
unsigned int scap_flags;

...

flags = bpf_syscall_get_argument(data, 2);
scap_flags = dup3_flags_to_scap(flags);
return bpf_push_u32_to_ring(data, scap_flags);
```

Also we seem to be missing those updates to the kernel module, aka `f_sys_dup3_x`.

As a result a lot of the current changes would need to be adjusted as well.

Reply:
Yes, you are right. I added the change to leave push to ring as unsigned. I also made the change in the driver for the flags to be represented as a signed integer. Let me know if I am missing something. Thanks!